### PR TITLE
feat: Container on product selection and settings screens [ROAD-449]

### PR DESCRIPTION
**Pull request by ArtsiomCh**
To enable use `snyk.preview.container.enabled` Idea Registry key:
image

image
image
image

### CODE REVIEW

File: src/integTest/kotlin/io/snyk/plugin/ui/toolwindow/settings/ScanTypesPanelTest.kt [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+        setUpContainerTest()
+        val settings = pluginSettings()
+        settings.containerScanEnabled = true
+
+        val scanTypesPanel = ScanTypesPanel(project, disposable)
+        val containerCheckBox =
+            getComponentByName(scanTypesPanel.panel, JBCheckBox::class, "containerEnablementCheckBox")
+                ?: throw IllegalStateException("containerEnablementCheckBox not found")
+
+        assertTrue("", containerCheckBox.isSelected)
```
Comment (bastiandoetsch) [inline src/integTest/kotlin/io/snyk/plugin/ui/toolwindow/settings/ScanTypesPanelTest.kt:61]: Should have an error message for the case your checking. Like "Expected container checkbox to be checked from setting, but wasn't"

File: src/integTest/kotlin/io/snyk/plugin/ui/toolwindow/settings/ScanTypesPanelTest.kt [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+        settings.containerScanEnabled = true
+        val scanTypesPanel = ScanTypesPanel(project, disposable)
+        val containerCheckBox =
+            getComponentByName(scanTypesPanel.panel, JBCheckBox::class, "containerEnablementCheckBox")
+                ?: throw IllegalStateException("containerEnablementCheckBox not found")
+
+        containerCheckBox.doClick()
+        scanTypesPanel.panel.apply()
+
+        assertFalse("", settings.containerScanEnabled)
```
Comment (bastiandoetsch) [inline src/integTest/kotlin/io/snyk/plugin/ui/toolwindow/settings/ScanTypesPanelTest.kt:77]: see above :)

File: src/integTest/kotlin/io/snyk/plugin/ui/toolwindow/settings/ScanTypesPanelTest.kt [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+        setUpContainerTest()
+        val settings = pluginSettings()
+        settings.containerScanEnabled = false
+
+        val scanTypesPanel = ScanTypesPanel(project, disposable)
+        val containerCheckBox =
+            getComponentByName(scanTypesPanel.panel, JBCheckBox::class, "containerEnablementCheckBox")
+                ?: throw IllegalStateException("containerEnablementCheckBox not found")
+
+        assertFalse("", containerCheckBox.isSelected)
```
[the remark this answers is not in the document]
Comment (bastiandoetsch) [reply src/integTest/kotlin/io/snyk/plugin/ui/toolwindow/settings/ScanTypesPanelTest.kt:91]: Is this really a different case than the first case? It is checking the same logic, if settings are forwarded to the checkbox.

File: src/integTest/kotlin/io/snyk/plugin/ui/toolwindow/settings/ScanTypesPanelTest.kt [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+        settings.containerScanEnabled = false
+        val scanTypesPanel = ScanTypesPanel(project, disposable)
+        val containerCheckBox =
+            getComponentByName(scanTypesPanel.panel, JBCheckBox::class, "containerEnablementCheckBox")
+                ?: throw IllegalStateException("containerEnablementCheckBox not found")
+
+        containerCheckBox.doClick()
+        scanTypesPanel.panel.apply()
+
+        assertTrue("", settings.containerScanEnabled)
```
Comment (bastiandoetsch) [inline src/integTest/kotlin/io/snyk/plugin/ui/toolwindow/settings/ScanTypesPanelTest.kt:107]: see above. Just for the sake of mentioning it - we do have some code duplication here, so one option would be to extract the common parts, a second option would be to use a parametrized test (see https://www.tutorialspoint.com/junit/junit_parameterized_test.htm)

File: src/main/kotlin/io/snyk/plugin/services/SnykApplicationSettingsStateService.kt [hunk 5]
```diff
@@ -37,7 +37,8 @@ class SnykApplicationSettingsStateService : PersistentStateComponent<SnykApplica
     var advisorEnable: Boolean = true
     var snykCodeSecurityIssuesScanEnable: Boolean = true
     var snykCodeQualityIssuesScanEnable: Boolean = false
-    var iacScanEnabled: Boolean = false
+    var iacScanEnabled: Boolean = true
```
Comment (bastiandoetsch) [inline src/main/kotlin/io/snyk/plugin/services/SnykApplicationSettingsStateService.kt:40]: Are we already enabling IaC with the next release?

File: src/integTest/kotlin/io/snyk/plugin/ui/toolwindow/settings/ScanTypesPanelTest.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (ArtsiomCh) [reply src/integTest/kotlin/io/snyk/plugin/ui/toolwindow/settings/ScanTypesPanelTest.kt:91]: Not really the same logic. The state of settings_field/checkbox (could) matter.

File: src/main/kotlin/io/snyk/plugin/services/SnykApplicationSettingsStateService.kt (hunk 5 above)
Comment (ArtsiomCh) [reply src/main/kotlin/io/snyk/plugin/services/SnykApplicationSettingsStateService.kt:40]: Enablement for the end users hidden by the feature flag.
Here we just change default enablement for IaC product according to Georgi's decision.

**Review by bastiandoetsch — APPROVED** [review]
LGTM

**Pull request merged**